core/src/jvmMain/kotlin/com/xebia/functional/xef/conversation/PlatformConversation.jvm.kt

```diff
@@ -58,12 +63,15 @@ actual constructor(
       }
       .asCompletableFuture()
 
-  fun chatFunction(target: Class<*>): CFunction =
-    CFunction(
+  fun chatFunction(target: Class<*>): CFunction {
+    val targetString = JacksonSerialization.schemaGenerator.generateSchema(target).toString()
+    return CFunction(
       name = target.simpleName,
       description = "Generated function for ${target.simpleName}",
-      parameters = JacksonSerialization.schemaGenerator.generateSchema(target).toString()
+      parameters = Json.parseToJsonElement(targetString).jsonObject
     )
```

Comment:
I guess this is ok because in this case we generate the schema specifically for JVM cases only when you provide a Class. This only affects the Java and Scala serializer and bypasses the kotlinx serialization. As an alternative, we would have to find a way that given a `Class` we can get back a JSON element from kotlin serialization and therefore avoid Jackson. 
The alternative is to implement a serialization stack for `Class` or find a library that already does this if we don't want to rely on Jackson.
```kotlin
@OptIn(InternalSerializationApi::class)
inline fun Class<*>.jsonSchema(): String =
  encodeJsonSchema(buildSerialDescriptor(
    serialName = this.simpleName,
    kind = if (isEnum) SerialKind.ENUM else SerialKind.CONTEXTUAL,
    typeParameters = emptyArray()
  ) {
    declaredFields.forEach { 
      element(it.name, ...)
    }
  })
```
I think your solution while it has to deserialize to serialize again to JsonObject is fine for now.